hazelcast-jet-core/src/main/java/com/hazelcast/jet/stream/impl/reducers/GroupingSinkReducer.java

```diff
@@ -16,52 +16,57 @@
 
 package com.hazelcast.jet.stream.impl.reducers;
 
-import com.hazelcast.core.IMap;
 import com.hazelcast.jet.DAG;
+import com.hazelcast.jet.Distributed.Function;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.ProcessorSupplier;
 import com.hazelcast.jet.Vertex;
-import com.hazelcast.jet.stream.DistributedCollector.Reducer;
+import com.hazelcast.jet.stream.DistributedCollector;
 import com.hazelcast.jet.stream.impl.pipeline.Pipeline;
 import com.hazelcast.jet.stream.impl.pipeline.StreamContext;
 import com.hazelcast.jet.stream.impl.processor.CombineGroupsP;
 import com.hazelcast.jet.stream.impl.processor.GroupAndAccumulateP;
 
-import java.util.function.Function;
 import java.util.stream.Collector;
 
 import static com.hazelcast.jet.Edge.between;
 import static com.hazelcast.jet.DistributedFunctions.entryKey;
 import static com.hazelcast.jet.Partitioner.HASH_CODE;
-import static com.hazelcast.jet.Processors.writeMap;
 import static com.hazelcast.jet.stream.impl.StreamUtil.executeJob;
 
-public class GroupingIMapReducer<T, A, K, D> implements Reducer<T, IMap<K, D>> {
+public class GroupingSinkReducer<T, A, K, D, R> implements DistributedCollector.Reducer<T, R> {
 
-    private final String mapName;
+    private final String name;
+    private final Function<JetInstance, ? extends R> toDistributedObject;
     private final Function<? super T, ? extends K> classifier;
     private final Collector<? super T, A, D> collector;
+    private final ProcessorSupplier processorSupplier;
 
-    public GroupingIMapReducer(String mapName, Function<? super T, ? extends K> classifier,
-                                Collector<? super T, A, D> collector) {
-        this.mapName = mapName;
+    public GroupingSinkReducer(String name,
```

Comment:
it would be better to rename name -> sinkName, and pass the full name i.e. "write-map-mapName"